python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -246,6 +246,40 @@ def test_append_index(a, b):
     assert_eq(expected.index, actual.index)
 
 
+@pytest.mark.parametrize(
+    "data",
+    [
+        {"a": [1, 2]},
+        {"a": [1, 2, 3], "b": [3, 4, 5]},
+        {"a": [1, 2, 3, 4], "b": [3, 4, 5, 6], "c": [1, 3, 5, 7]},
+        {"a": [np.nan, 2, 3, 4], "b": [3, 4, np.nan, 6], "c": [1, 3, 5, 7]},
+        {1: [1, 2, 3], 2: [3, 4, 5]},
+        {"a": [1, None, None], "b": [3, np.nan, np.nan]},
+        {1: ["a", "b", "c"], 2: ["q", "w", "u"]},
+        {1: ["a", np.nan, "c"], 2: ["q", None, "u"]},
+    ],
+)
+def test_axes_dataframe(data):
+    csr = cudf.DataFrame(data)
+    psr = csr.to_pandas()
+
+    expected = psr.axes
+    actual = csr.axes
+
+    for i in range(len(actual)):
+        assert_eq(expected[i], actual[i])
+
+    # When the DataFrame is empty and when there are column names
+    # The expected output is slight different compared to logic
+    # Expected - return [Index([],dtype = 'object'), self.columns]
+    # Actual - return [self.index, self.columns]
+
+    # When the DataFrame is not empty, and column names are not available
+    # The expected output is slightly different compared to logic
+    # Expected - return [self.index, RangeIndex(0,1)]
+    # Actual - return [self.index, self.columns]
```

Comment:
I'm having a little trouble following this comment, would you be able to help me with some examples? If possible it's worth trying to match pandas in various edge cases involving empty dataframes, it avoids creating subtle bugs when users try and convert pandas workflows to cuDF.

Reply:
>>> import cudf
>>> import pandas as pd

>>> cdf = cudf.DataFrame([1,2,3])
>>> pdf = pd.DataFrame([1,2,3])

>>> pdf.axes
[RangeIndex(start=0, stop=3, step=1), RangeIndex(start=0, stop=1, step=1)]

>>> cdf.axes
[RangeIndex(start=0, stop=3, step=1), Int64Index([0], dtype='int64')]

Reply:
>>> cdf = cudf.DataFrame(columns =[1,2,3])
>>> pdf = pd.DataFrame(columns = [1,2,3])
>>> pdf.axes
[Index([], dtype='object'), Int64Index([1, 2, 3], dtype='int64')]

>>> cdf.axes
[RangeIndex(start=0, stop=0, step=1), Int64Index([1, 2, 3], dtype='int64')]

Reply:
I think the first case is happening because of a subtle bug with `to_pandas_index()`. We have
```python
>>> pd.DataFrame([1,2,3]).columns
RangeIndex(start=0, stop=1, step=1)
>>> cudf.DataFrame([1,2,3]).to_pandas().columns
Int64Index([0], dtype='int64')
```

This isn't right and leads to the bug here where the behavior is different in this edge case. We can do one of these things:

1. Fix the bug as part of this PR, adding tests
2.  Do not fix the bug in this PR, but add tests for the edge cases, opening an issue and linking to it in the `xfail(reason=...)`.
3. Open an issue about the bug in `to_pandas_index`, merge a PR fixing it, then add the edge cases to the tests here and merge this PR once those pass.

Overall I think (3) is the cleanest and involves the least amount of writing and removing code later. But the other options are fine with me too.